pkg/kubeapiserver/options/authentication_test.go

```diff
@@ -265,6 +266,9 @@ func TestAuthenticationValidate(t *testing.T) {
 				featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, f, true)
 			}
 			for _, f := range testcase.disabledFeatures {
```

Comment:
SetFeatureGateEmulationVersionDuringTest needs to be called before any calls to SetFeatureGateDuringTest to work reliably

add a new loop at the top of the test to just do this bit:
```go
for _, f := range testcase.disabledFeatures {
  if f == features.AnonymousAuthConfigurableEndpoints {
    featuregatetesting.SetFeatureGateEmulationVersionDuringTest(t, utilfeature.DefaultFeatureGate, version.MustParse("1.33"))
  }
}
```